File: frontend/src/lib/hosts.ts
export const defaultHostId = "featured-host-family";

export const hosts = [
  {
    id: "featured-host-family",
    label: "Featured Host Family",
    available: true,
    image: "/images/host-home-entrance.jpeg",
  },
  {
    id: "future-host-family",
    label: "Future Host Family",
    available: false,
    image: "/images/family-living-dining-room.jpeg",
  },
  {
    id: "seasonal-host-family",
    label: "Seasonal Host Family",
    available: false,
    image: "/images/riverside-sunset.jpeg",
  },
] as const;

export type HostId = (typeof hosts)[number]["id"];

export function isHostId(value: string): value is HostId {
  return hosts.some((host) => host.id === value);
}

export function hostById(id: string) {
  return hosts.find((host) => host.id === id);
}

File: backend/app/bookings/hosts.py
from pydantic import BaseModel


class HostDefinition(BaseModel):
    id: str
    name: str
    bookable: bool


HOSTS = {
    "featured-host-family": HostDefinition(
        id="featured-host-family",
        name="Featured Host Family",
        bookable=True,
    ),
    "future-host-family": HostDefinition(
        id="future-host-family",
        name="Future Host Family",
        bookable=False,
    ),
    "seasonal-host-family": HostDefinition(
        id="seasonal-host-family",
        name="Seasonal Host Family",
        bookable=False,
    ),
}


def host_for_id(host_id: str) -> HostDefinition | None:
    return HOSTS.get(host_id)
